### PR TITLE
DVCSMP-2134, DVCSMP-2135 ZLL dimmable and white color temp bulb health check

### PR DESCRIPTION
**Pull request by surfous**
- configure a healthPoll every 5 min
  - dimmable refreshes on/off state and switch level (DVCSMP-2135)
  - white color temperature refreshes on/off state, switch level, and white color temperature (DVCSMP-2134)
  - set up checkInterval from both configure and updated
- checkInterval set to 12 min (2 misses + leeway)
- ping refreshes level setting

### CODE REVIEW

**Comment by surfous** [comment]
@workingmonk and/or @jackchi, please code review or suggest other(s) who should review. 

I will ping Lakshmy to arrange QA for these two handlers once it passes code review.

**Review by workingmonk — APPROVED** [review]
looks good. but also get @jackchi  to approve from a health pov

File: devicetypes/smartthings/zll-dimmer-bulb.src/zll-dimmer-bulb.groovy
Comment (jackchi) [inline devicetypes/smartthings/zll-dimmer-bulb.src/zll-dimmer-bulb.groovy]: I see you added `configureHealthCheck()` to both configure() and updated(). Is this necessary? What was the thought behind this? Thanks

Comment (surfous) [reply devicetypes/smartthings/zll-dimmer-bulb.src/zll-dimmer-bulb.groovy]: Necessary? I'm not sure. This was a pattern I'd seen in the docs, and now that I look, it was  used in SmartApps calling a common initialize() method from both configure() and updated():
http://docs.smartthings.com/en/latest/smartapp-developers-guide/anatomy-and-life-cycle-of-a-smartapp.html#smartapp-structure

Looking at your comment on CHF-429, it seems that updated() is called if the DH for a device is changed in the IDE. What about in the case of deploying a new version of the **same** DH which changes the checkInterval?

Regardless, the best answer I can give is another question: For a DH which implements capability configurable, can configure() ever be invoked in an execution where updated() would not be?

If the answer is a definitive no, then there's no need to call it from both places.

Finally, it seems that I should leave the ZigBee.onOffConfig()... line with configure() for these DHs as that's configuring the device, which is the intent of the capability.configurable - do you agree?

Comment (jackchi) [reply devicetypes/smartthings/zll-dimmer-bulb.src/zll-dimmer-bulb.groovy]: After talking to you this is the pattern we want to go with. Looks good! 👍

Comment (surfous) [reply devicetypes/smartthings/zll-dimmer-bulb.src/zll-dimmer-bulb.groovy]: Great - my subsequent changes implemented running configureHealthCheck() only once-per-execution as we'd discussed before merging on your approval.

**Comment by jackchi** [comment]
👍 
Let's have QA confirm all the supported models.

**Comment by jackchi** [comment]
@surfous I tried the Field variable but it doesn't seem to work as expected here. 
The difference between your script and the one I tried is that I put the `configureHealthCheck` in the `updated()` lifecycle method.

**Comment by surfous** [comment]
@jackchi How is it working that seems inconsistent with expectations? I would like to understand what it's doing or not doing.

**Comment by jackchi** [comment]
By adding this the Field variable, my `configureHealthCheck` ran 3 times when I put it in the `updated()` method. Also Morten pointed out that there isn't a good way of knowing the execution lifecycle of a DT / SA. Usually this is used as scripts. Also there isn't harm in sending the `checkInterval` multiple times.

**Comment by surfous** [comment]
@jackchi I understand what I need to understand here. 

Also, Field is working exactly as expected ;) We can talk about that later if you want - it's a solid tool. But you're right, since sending the `checkInterval` event more than once isn't detrimental, there's no need to use it here.

**Pull request merged**
